packages/cyberstorm/src/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,86 @@
+import { Fragment, FunctionComponent, ReactNode } from "react";
+
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { IndexLink } from "./Links";
+
+type Label = ReactNode | string;
+
+interface Crumb {
+  label: Label;
+  LinkComponent?: FunctionComponent;
+}
+
+type LinkCrumb<P> = Omit<Crumb, "LinkComponent"> & {
+  label: Label;
+  LinkComponent: FunctionComponent<P>;
+  LinkProps: P;
+};
+
+interface BreadCrumbProps<P> {
+  parts: (Crumb | LinkCrumb<P>)[];
+}
+
+export function BreadCrumb<P>(props: BreadCrumbProps<P>) {
+  if (!props.parts) {
+    return null;
+  }
+  const parts = [...props.parts];
+  const leaf = parts.pop();
+
+  return (
+    <div className={styles.root}>
+      {parts.map((crumb, i) => (
+        <Fragment key={`${crumb.label}-${i}`}>
+          <MidCrumb crumb={crumb} />
+          <Separator />
+        </Fragment>
+      ))}
+
+      <LeafCrumb crumb={leaf} />
+    </div>
+  );
+}
+
+function MidCrumb<P>(props: { crumb: Crumb | LinkCrumb<P> }) {
+  const { crumb } = props;
+  if (crumb.LinkComponent === undefined) {
+    return CrumbLabel(crumb.label);
+  }
+  if (!("LinkProps" in crumb)) {
```

Comment:
I guess this works here, but for future reference checkout [user-defined type guards](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). I prefer them since they're reusable, more readable and usually make the intent more clear.

Not a blocker.

Reply:
I've implemented something like what was in the example of "user-defined type guards".